public/javascripts/timeline.js

```diff
@@ -43,35 +50,115 @@ function update () {
     this.now_offset = .25 * hour_tick_list.offsetWidth + ((now - selectedDate) / MS_PER_HOUR ) * this.hour_width;
 
 
-    // Calculate the start point of the first hour
-    var hours_to_left = this.now_offset / this.hour_width;
+    // hours being displayed
+    if (slider_value > 3 && slider_value <= 8) {
+
+            // Determine the pixel offset of the current-time-bar from the left
+            this.now_offset = .25 * hour_tick_list.offsetWidth + ((now - selectedDate) / MS_PER_HOUR ) * hour_width;
+            document.querySelector('#current-time-bar').style.left = this.now_offset + "px"
+
+            // Calculate the start point of the first hour
+            var hours_to_left = this.now_offset / this.hour_width;
+
+            var hour_start = Math.floor(now.getHours() - hours_to_left);
+            this.initial_offset = now_offset + ((hour2Date(hour_start) - now) / MS_PER_HOUR) * this.hour_width;
+            var hour_tick = createHourTick(hour_start);
+     
+            hour_tick.style.left = initial_offset + "px";
+            hour_tick_list.appendChild(hour_tick);
+
+        for (var i = 1; i < num_hours + 2; i++) {
+            var hour_tick = createHourTick(i + hour_start);
+            hour_tick.style.left = initial_offset + "px";
+            hour_tick_list.appendChild(hour_tick);
+        }
+
+        // console.log(hour_tick_list)
+
+        updateElemOffset('.item');
+        updateEventDuration();
+    
+    // every fifteenth minutes and hours being displayed
+
+    } else if (slider_value <=3 && slider_value >=2) {
+
+             // Determine the pixel offset of the current-time-bar from the left
+            this.now_offset = .25 * hour_tick_list.offsetWidth + ((now - selectedDate) / MS_PER_HOUR ) * hour_width;
+            document.querySelector('#current-time-bar').style.left = this.now_offset + "px"
+
+            // Calculate the start point of the first hour
+            var hours_to_left = this.now_offset / this.hour_width;
+
+            var hour_start = Math.floor(now.getHours() - hours_to_left);
+            this.initial_offset = now_offset + ((hour2Date(hour_start) - now) / MS_PER_HOUR) * this.hour_width;
+            var hour_tick = createHourTick2(hour_start);
+     
+            hour_tick.style.left = initial_offset + "px";
+            hour_tick_list.appendChild(hour_tick);
+
+        for (var i = 1; i < num_hours + 1; i++) {
+            var hour_tick = createHourTick2(i + hour_start);
+            hour_tick.style.left = initial_offset + "px";
+            hour_tick_list.appendChild(hour_tick);
+        }
+
+        // console.log(hour_tick_list)
+
+        updateElemOffset('.item');
+        updateEventDuration();
+
+        // display minutes and hours
+
```

Comment:
I'm having trouble finding what you have changed above to get the 15 minute intervals. Would you mind commenting the above to pick out the changes?